src/osdc/Objecter.cc

```diff
@@ -1944,7 +1944,7 @@ void Objecter::_maybe_request_map()
     flag = CEPH_SUBSCRIBE_ONETIME;
   }
   epoch_t epoch = osdmap->get_epoch() ? osdmap->get_epoch()+1 : 0;
-  if (monc->sub_want("osdmap", epoch, flag)) {
+  if (monc->sub_want("osdmap", epoch, flag|CEPH_SUBSCRIBE_LATEST_OSDMAP)) {
```

Comment:
Are you sure all Objecter users would want this?

Tying to the "noosd" vs "osd" comment above, Objecter is also used by managers, MDSes and even OSDs in some cases, I think.  If the intention is to turn on `CEPH_SUBSCRIBE_LATEST_OSDMAP` behaviour just for clients, this is probably wrong.

Reply:
Nit: need spaces on each side of |.

Reply:
I search osdmap and want_sub and find that this function, want_sub , is called by only osd and  objecter.
I will change it about issue of spaces.

Reply:
updated it